mne/epochs.py

```diff
@@ -3249,6 +3260,38 @@ def __init__(
         # keep track of original sfreq (needed for annotations)
         raw_sfreq = raw.info["sfreq"]
 
+        # get events from annotations if no events given
+        if events is None:
```

Comment:
I would put this block in a private function to be clear about what it needs and returns

Reply:
+1, the `__init__` is already long enough as it is and a separate function would help with that